File: kotlin/AlternateCase.kt
fun alternateCase(input: String): String {
    var makeUpper = true
    val result = StringBuilder()
    for (ch in input) {
        result.append(
            when {
                ch.isLetter() -> {
                    val converted = if (makeUpper) ch.uppercaseChar() else ch.lowercaseChar()
                    makeUpper = !makeUpper
                    converted
                }
                else -> ch
            }
        )
    }
    return result.toString()
}

fun main() {
    println(alternateCase("instagram"))  // InStAgRaM
    println(alternateCase("hello world"))  // HeLlO wOrLd
    println(alternateCase("abc123def"))  // AbC123DeF
}

File: csharp/AlternateCase.cs
using System;
using System.Text;

class AlternateCase
{
    static string ToAlternateCase(string input)
    {
        bool makeUpper = true;
        StringBuilder result = new StringBuilder();
        foreach (char ch in input)
        {
            if (char.IsLetter(ch))
            {
                result.Append(makeUpper ? char.ToUpper(ch) : char.ToLower(ch));
                makeUpper = !makeUpper;
            }
            else
            {
                result.Append(ch);
            }
        }
        return result.ToString();
    }

    static void Main()
    {
        Console.WriteLine(ToAlternateCase("instagram"));   // InStAgRaM
        Console.WriteLine(ToAlternateCase("hello world")); // HeLlO wOrLd
        Console.WriteLine(ToAlternateCase("abc123def"));   // AbC123DeF
    }
}
